service/authorization/v2/authorization.proto

```diff
@@ -0,0 +1,224 @@
+syntax = "proto3";
+
+package authorization.v2;
+
+import "authorization/authorization.proto";
+import "buf/validate/validate.proto";
+import "google/protobuf/any.proto";
+import "google/protobuf/wrappers.proto";
+import "policy/objects.proto";
+
+enum Decision {
+  DECISION_UNSPECIFIED = 0;
+  DECISION_DENY = 1;
+  DECISION_PERMIT = 2;
+  // DECISION_OBLIGATED = 3;
+}
+
+// PE (Person Entity) or NPE (Non-Person Entity)
+message Entity {
+  // ephemeral id for tracking between request and response
+  string ephemeral_id = 1;
+  // Standard entity types supported by the platform
+  oneof entity_type {
+    // one of the entity options must be set
+    string email_address = 2;
+    string user_name = 3;
+    // used by ERS claims mode
+    google.protobuf.Any claims = 4;
+    // EntityCustom custom = 5;
+    string client_id = 6;
+    string registered_resource_value_fqn = 7;
+  }
+  enum Category {
+    CATEGORY_UNSPECIFIED = 0;
+    CATEGORY_SUBJECT = 1;
+    CATEGORY_ENVIRONMENT = 2;
+  }
+  Category category = 9;
+}
+
+// Entity type for custom entities beyond the standard types
+// message EntityCustom {
+//   google.protobuf.Any extension = 1;
+// }
+
+// Entitlements for a given entity, mapping each attribute value FQN to any entitled actions[]
+message EntityEntitlements {
+  message ActionsList {
+    repeated policy.Action actions = 1;
+  }
+
+  // ephemeral id for tracking between request and response
+  string ephemeral_id = 1;
+  map<string, ActionsList> actions_per_attribute_value_fqn = 2;
+}
+
+// Either a set of attribute values (such as those on a TDF) or a registered resource value
+message Resource {
+  // ephemeral id for tracking between request and response
+  string ephemeral_id = 1;
+
+  message AttributeValues {
+    repeated string fqns = 1;
+  }
+
+  oneof resource {
+    AttributeValues attribute_values = 2;
+    string registered_resource_value_fqn = 3;
+  }
+}
+
+message ResourceDecision {
+  // ephemeral id for tracking between request and response
+  string ephemeral_resource_id = 1;
+  // the decision result
+  Decision decision = 3;
+}
+
+// Can the entity (PE or NPE) access?
+// 1. one entity chain
+// 2. one action
+// 3. one resource
+message GetDecisionRequest {
+  authorization.EntityChain entity = 1 [(buf.validate.field).required = true];
+  // name on action is required
+  policy.Action action = 2 [(buf.validate.field).required = true];
+  Resource resource = 3 [(buf.validate.field).required = true];
```

Comment:
if a decision request only has a single resource, do we need the ephemeral_resource_id in the ResourceDecision on line 74 since theres only one? same thing for the entity chain ephemeral id in the decision response i guess - if were only doing a 1x1 evaluation do we need ids?

Reply:
needed for multi-resource decisioning, and bulk RPCs